server/src/server.js

```diff
@@ -645,6 +645,14 @@ app.get("/images/:imageid", function (req, res) {
         anonAnalytics.pageview(analyticsUrl).send();
       }
       res.header("Content-Type", obj.contentType);
+      if (req.query.filename) {
+        let extension = "." + obj.contentType.split("/").pop();
+        let filename= req.query.filename;
+        if (! filename.endsWith(extension)) {
+          filename = filename + extension;
+        }
+        res.header("Content-Disposition", `attachment; filename="${filename}"`)
+      }
```

Comment:
You have to pass the generated URL through the model.

The download attribute works like  `<a href="uuid-link" download="filename">` with `download` as a suggestion, and the browser handles making sure it is cleaned up.  So it's simpler and only requires switching to an `<a>` tag, which is more semantic anyway.